neuraxle/metaopt/auto_ml.py

```diff
@@ -465,12 +476,9 @@ def fit(self, p, train_data_container: DataContainer, validation_data_container:
 
         for i in range(self.epochs):
             self.print_func('\nepoch {}/{}'.format(i + 1, self.epochs))
-            p = p.handle_fit(train_data_container, context)
-
-            y_pred_train = p.handle_predict(train_data_container, context)
-            y_pred_val = p.handle_predict(validation_data_container, context)
-
-            trial.set_fitted_pipeline(pipeline=p)
+            trial = trial.fit(train_data_container, context)
+            y_pred_train = trial.predict(train_data_container, context)
+            y_pred_val = trial.predict(validation_data_container, context)
```

Comment:
here is what I was talking about with trial.fit @guillaume-chevalier

Reply:
would rename method maybe (see other comment, resolve her eanytime)